### ISSUE TITLE
"Role" property of conversation update activity set to null

### ISSUE THREAD

**Issue by manish-95** [microsoft/botframework-solutions#3614]
## Version
WebChat 4.9.1 via CDN

## Describe the bug

When the conversationUpdate activity is sent by WebChat, the Role property of the activity is set to null. As a result, the ProactiveState does not get updated with new Conversation details since it looks for the role being equal to "user". This gets updated only when a skill is invoked/message is sent. As a result, if there are any proactive notifications, they are sent to the wrong conversation.

## Steps to reproduce

1. Initialize connection to Bot through WebChat using DirectLine.
2. Open a connection to the Bot.
3. Note in the bot back-end logic that turnContext.Activity.From.Role is set to null, and not "user".
4. Note the error.

## Expected behavior

WebChat should pass the relevant role as "user", so that the Proactive State functionalities are not broken.

## Additional context

WebChat implemented in Angular application.

[Bug]

**Comment by peterinnesmsft** [comment]
@ryanlengel @lzc850612 could you take a look and do an initial pass of this issue?

**Comment by lzc850612** [comment]
@EricDahlvang the reason we were using activity.from.role is because in the middleware we need to only store the conversation reference of the activity that's coming from user to bot. if we don't use activity.from.role, what other way we could use to achieve this validation?


Thanks.

**Comment by EricDahlvang** [comment]
@lzc850612 I'm not sure.  Is ProactiveStateMiddleware.OnTurn ever executed with an activity which is NOT from a user?  If so, maybe a custom property?  

Either way, ProactiveStateMiddleware should be consistent with whatever is done in DefaultActivityHandler ... this inconsistency is the cause of the issue right now.

**Comment by peterinnesmsft** [comment]
@manish-95, @lzc850612 has checked in a PR related to this which we believe should help address this issue. We will work on actively pushing new releases in the next week or two. In the meantime, can you please take a look and work with him to see if this helps resolve your issue? Thanks!

**Comment by manish-95** [comment]
> @manish-95, @lzc850612 has checked in a PR related to this which we believe should help address this issue. We will work on actively pushing new releases in the next week or two. In the meantime, can you please take a look and work with him to see if this helps resolve your issue? Thanks!

@peterinnesmsft @lzc850612  Sure, I'll definitely do that. For now , I have added a dirty fix for the same, by adding the Role property for each activity, but I can quickly copy the contents from this PR locally and verify.

**Comment by peterinnesmsft** [comment]
@manish-95 any luck on verifying this? We are tentatively looking to do some initial validation of a release candidate containing this fix next week, but any heads up if you've caught obvious issues would be greatly appreciated if you have them!

**Comment by manish-95** [comment]
Hi @peterinnesmsft , my apologies for the extremely late update on this, however, I have taken a copy of the file with the new changes, and it seems to work fine, for all the cases that I am using.

**Comment by peterinnesmsft** [comment]
@manish-95 thank you for confirming!

@Batta32 as I expect this will be resolved by our updates to the SDK package references, I'll assign this to you for further tracking. Once we have pushed out our own updates with the updated SDK references and validated, let's go ahead and close this out.

**Comment by mrivera-ms** [comment]
@peterinnesmsft can this be closed now?

**Comment by EricDahlvang** [comment]
Has this been tested against the latest sdk?  The sdk is now utilizing the activity.recipient.role property.  When a ConversationReference for a continue call, is created, this role will go with it.   I'm guessing there is will now be conflicts with how DefaultActivityHandler.cs and ProactiveStateMiddleware.cs are using .Role

**Comment by Batta32** [comment]
Hi @mrivera-ms and @EricDahlvang , we already **validated** that this is correctly fixed with SDK R11 and we created the PR #3707 in order to use the latest SDK version.

As soon as the PR #3707 is merged and new packages are published with this new version, this will be automatically solved.

![image](https://user-images.githubusercontent.com/11904023/99425282-660fdb80-28e1-11eb-9241-f209b31fc702.png)

**Comment by EricDahlvang** [comment]
@Batta32 have proactive skills been tested with 4.11?  the code paths which utilize DefaultActivityHandler.cs and ProactiveStateMiddleware.cs ?

**Comment by Batta32** [comment]
Sorry @EricDahlvang, we meant that we successfully validated this using C# Bot-Solutions version [1.0.1-daily322](https://botbuilder.myget.org/feed/aitemplates/package/nuget/Microsoft.Bot.Solutions/1.0.1-daily322) (`aitemplates` feed) which contains the PR #3633 and uses SDK R9.

Should we validate this using R11?

**Comment by hcyang** [comment]
@EricDahlvang, can you respond to @Batta32's comment? Thanks.

**Comment by EricDahlvang** [comment]
Sorry, I'm not sure.  Looks like maybe this will not be an issue, since #3633 removed the check for role?

**Comment by Batta32** [comment]
@EricDahlvang - PR #3633 removed the check for role.

Just in case, we will do a validation of R11 with the changes implemented in PR #3633. As soon as we have any update, we will let you know the result 😊.

**Comment by Batta32** [comment]
Hi @EricDahlvang, sorry for the delay. We validated the fix of PR #3633 which removes the check for role using R11 and we confirmed that it's **working** having the `role` property set to null using Web Chat and DirectLine.

We think that this issue can be closed as the changes will be present in the new stable packages of Solutions.

![image](https://user-images.githubusercontent.com/11904023/101682831-e938d300-3a42-11eb-870a-1eac688096a0.png)
